selection_sort.go

```diff
@@ -0,0 +1,16 @@
+package sort
+
+func selectionSort(arr []int) []int {
+	for i := 0; i < len(arr)-1; i++ {
+		minIdx := i
+		for j := i; j < len(arr); j++ {
+			if arr[minIdx] > arr[j] {
+				minIdx = j
+			}
+		}
+		if minIdx != i {
```

Comment:
The check `if minIdx != i` is correct, but consider adding a comment to explain why this check is necessary.